src/NServiceBus.NHibernate/SynchronizedStorage/NHibernateLazyNativeTransactionSynchronizedStorageSession.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace NServiceBus.Persistence.NHibernate
+{
+    using System;
+    using System.Threading.Tasks;
+    using global::NHibernate;
+    using Janitor;
+
+    [SkipWeaving]
+    class NHibernateLazyNativeTransactionSynchronizedStorageSession : CompletableSynchronizedStorageSession, INHibernateSynchronizedStorageSession
+    {
+        Func<ISession> sessionFactory;
+        ISession session;
+        ITransaction transaction;
+
+        public NHibernateLazyNativeTransactionSynchronizedStorageSession(Func<ISession> sessionFactory)
+        {
+            this.sessionFactory = sessionFactory;
+        }
+
+        public ISession Session
+        {
+            get
+            {
+                if (session == null)
```

Comment:
`Lazy` is thread safe but we don't need thread safety here. Do you still think it would be read better? If so, I can change it. No strong opinions.